tracer/equation.py

```diff
@@ -198,22 +223,29 @@ class ConservativeTracerEquation2D(Equation):
     """
     def __init__(self, function_space, depth,
                  use_lax_friedrichs=False,
-                 sipg_parameter=Constant(10.0)):
+                 sipg_parameter=Constant(10.0),
+                 anisotropic=False):
         """
         :arg function_space: :class:`FunctionSpace` where the solution belongs
         :arg depth: :class: `DepthExpression` containing depth info
         :kwarg bool use_lax_friedrichs: whether to use Lax Friedrichs stabilisation
         :kwarg sipg_parameter: :class: `Constant` or :class: `Function` penalty parameter for SIPG
+        :kwarg anisotropic: toggle anisotropic cell size measure
         """
-        super(ConservativeTracerEquation2D, self).__init__(function_space)
-        args = (function_space, depth, use_lax_friedrichs, sipg_parameter)
-        self.add_term(ConservativeHorizontalAdvectionTerm(*args), 'explicit')
-        self.add_term(ConservativeHorizontalDiffusionTerm(*args), 'explicit')
+        super(ConservativeTracerEquation2D, self).__init__(function_space, anisotropic=anisotropic)
+        args = (function_space, depth)
+        kwargs = {
+            'use_lax_friedrichs': use_lax_friedrichs,
+            'sipg_parameter': sipg_parameter,
+        }
+        self.add_term(ConservativeHorizontalAdvectionTerm(*args, **kwargs), 'explicit')
+        self.add_term(ConservativeHorizontalDiffusionTerm(*args, **kwargs), 'explicit')
         if self.function_space.ufl_element().family() == 'Lagrange':
-            self.add_term(SourceTerm(*args), 'source')
+            self.add_term(SourceTerm(*args, **kwargs), 'source')
         else:
```

Comment:
I don't quite understand how using a lagrange element is linked to using a conservative or non-conservative source term

Reply:
oh I see makes sense!

Reply:
Fixed now. Note that the formulation for the CG conservative equation is slightly different from DG. (Because of a use-case I am looking at.)